pixy/httpapi.go

```diff
@@ -131,3 +143,30 @@ func getParamBytes(r *http.Request, name string) []byte {
 	}
 	return []byte(values[0])
 }
+
+func replyBadRequest(w http.ResponseWriter, format string, a ...interface{}) {
+	w.WriteHeader(http.StatusBadRequest)
+	w.Write([]byte(fmt.Sprintf(format, a...)))
+}
+
+// readBody reads from r until an error or EOF and returns the data it read
+// from the internal buffer allocated with a specified capacity.
+func readBody(r io.ReadCloser, capacity int64) (b []byte, err error) {
```

Comment:
The point of readBody is to allocate all the memory needed to accommodate `Content-Length` bytes. The solution above still can make several allocation. I kind of wished `ioutil.ReadAll` function had **capacity** parameter. On the other hand worrying about memory allocation in this case is probably premature optimization and I should ditch extra logic (readBody) function in faivour of standard solution (ioutil.ReadAll)